client/src/components/Room.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import styled from 'styled-components'
+import { useStore } from 'outstated'
+import { useRoomStore as roomStore } from 'lib/stores'
+import { VideoPlayer } from './VideoPlayer'
+import { PeerState } from 'lib/stores/useRoomStore'
+
+export const Room: React.FunctionComponent = () => {
+    const { id } = useParams()
+    const { ws, peer, stream, peers } = useStore(roomStore)
+
+    useEffect(() => {
+        if (peer) {
+            // eslint-disable-next-line no-underscore-dangle
+            ws.emit('join-room', { roomId: id, peerId: peer._id })
+        }
+    }, [id, peer, ws])
+
+    return (
+        <VideoContainer>
+            <VideoPlayer stream={stream}/>
+            {Object.values(peers as PeerState).map((peer, index) => (
```

Comment:
this casting should not take place here, instead of this type the peers in the store